### DISCUSSION TITLE
add multiple options?

### DISCUSSION THREAD

**Discussion by Cryptodarktrader** [SilentVoid13/Templater#1229] [Help]
Hello, I wanted to ask you a question because no matter how much I try, I can't find the solution. I have this script for a cooking recipe note:

```
<%* const category = await tp.system.suggester([
"🥣Soup",
"❄️Cold",
"💚Vegan",
"🥬Vegetarian",
"🥗Salad (Ensalada)",
"🍤Seafood (Marisco)",
"🧁Dessert (Postres)",
"🍰Cake (Tarta)",
"🍗Chicken (Pollo)",
"🥩Meat (Carne)",
"🦆Duck (Pato)",
"🐷Pork (Cerdo)",
"🐮Beef (ternera)",
"🍛Stew (Guiso)",
"🍚Rice (Arroz)",
"🌶️Spicy (Picante)"
], [
"🥣Soup",
"❄️Cold",
"💚Vegan",
"🥬Vegetarian",
"🥗Salad",
"🍤Seafood",
"🧁Dessert",
"🧁Dessert/🍰Cake",
"🍗Chicken",
"🥩Meat",
"🥩Meat/🦆Duck",
"🥩Meat/🐷Pork",
"🥩Meat/🐮Beef",
"🍛Stew",
"🍚Rice",
"🌶️Spicy"
], false, "¿Qué tipo de Receta es?") %>
```

I would need to find a way to ask me after choosing an option if I want to add another, and renew the same code and add that second, third or fourth option

I understand that the correct thing would be an if,  and to  new ask and depending on that answer, the code again or not, but I can't get it to work

**Comment by Cryptodarktrader** [comment]
ok I think I have solved it I have all this code in a note called scr.categories which I invoke with:
   <% tp.file.include("[[scr.categories]]") %>
from any note, it adds an option and asks me if I want to add another one, if so it places it separated with "," and it relaunches the code again and if I choose no it doesn't add anything.

It is perfectly modifiable for any other use in case it helps someone.
```
<%* const category = await tp.system.suggester([
"🥣Soup",
"❄️Cold",
"💚Vegan",
"🥬Vegetarian",
"🥗Salad",
"🍤Seafood",
"🧁Dessert",
"🧁Dessert/🍰Cake",
"🍗Chicken",
"🥩Meat",
"🥩Meat/🦆Duck",
"🥩Meat/🐷Pork",
"🥩Meat/🐮Beef",
"🍛Stew",
"🍚Rice",
"🌶️Spicy"
], [
"🥣Soup",
"❄️Cold",
"💚Vegan",
"🥬Vegetarian",
"🥗Salad",
"🍤Seafood",
"🧁Dessert",
"🧁Dessert/🍰Cake",
"🍗Chicken",
"🥩Meat",
"🥩Meat/🦆Duck",
"🥩Meat/🐷Pork",
"🥩Meat/🐮Beef",
"🍛Stew",
"🍚Rice",
"🌶️Spicy"
], false, "What type of Recipe is?") %><%* const question = await tp.system.suggester([
    "✅ Yes", 
    "⛔ No" 
], [
    "✅ Yes", 
    "⛔ No" 
], false, "Choose an option:") %><%* if (question === "✅ Yes") { %> 
<% tp.file.include("[[scr.categories]]") %>
<%* } else { %><%* } %>
```

What I haven't been able to correct yet and I don't know what causes it is that after the code is executed, it leaves a very large space in the properties:
![image](https://github.com/SilentVoid13/Templater/assets/145910014/5baaa575-da24-4792-9f3f-5132c160e5b7)

**Reply by Zachatoo** [reply]
You can use [whitespace control](https://silentvoid13.github.io/Templater/commands/whitespace-control.html) to fix the blank lines

**Reply by Cryptodarktrader** [reply]
God bless you, haha it was as easy as replacing the `-` in this line with a `_`:
```
], false, "Choose an option:") %><%* if (question === "✅ Yes") { %> 
<% tp.file.include("[[scr.categories]]") %>
<%* } else { %><%* } _%>
```


 and not only does it return the result without spaces but it also allows me to invoke it in another part of the template perfectly
Currently I have my recipe template with properties like this:
```
<%*
const title = await tp.system.prompt("Recipe Name?");
const categorie = await tp.file.include("[[scr.categories]]");
await tp.file.move("/20 - 🍜 Recipes/" + title);
-%>---
tags: 
title: <% title %>
aliases: 
cuisine: 
categories: <% categorie %>
servings: 
prep: 
Cook: 
total_time:
---

```
and further down there was a:

```
||
|:--:|
|<% categorie %>|
||
```
but it always gave me an error because it was invoked before or at the same time as the script, now it is invoked later so it copies the result and it looks perfectly

**Reply by Cryptodarktrader** [reply]
I attach the 2 notes in case anyone is interested, the template that can be invoked with templater and the script that I have saved in another folder (although it is also a note)
[tp.recipes.md](https://github.com/SilentVoid13/Templater/files/13071365/tp.recipes.md)
[scr.categories.md](https://github.com/SilentVoid13/Templater/files/13071366/scr.categories.md)
